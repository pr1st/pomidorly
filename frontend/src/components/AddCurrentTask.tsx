import React from 'react';
import classNames from "classnames";
import CurrentTaskParam from "./CurrentTaskParam";

type Props = { addTask: (tag: string, description: string, numberOfPomidors: number) => void }
type State = {
    tag: string,
    description: string,
    numberOfPomidors: number
}

const initState = {
    tag: "",
    description: "",
    numberOfPomidors: 1
};

class AddCurrentTask extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = initState
    }

    setParam = (paramName: string, value: number | string) => {
        if (paramName === "numberOfPomidors") {
            this.setState({
                ...this.state,
                numberOfPomidors: (value as number).valueOf()
            })
        } else {
            this.setState({
                    ...this.state,
                    [paramName]: (value as string).valueOf()
                }
            )
        }
    };

    add = () => {
        const {numberOfPomidors, description, tag} = this.state;
        this.props.addTask(tag, description, numberOfPomidors);
        this.setState({
            ...initState
        })
    };

    render() {
        const {numberOfPomidors, description, tag} = this.state

        return (
            <div className={classNames("row", "AddCurrentTask")}>
                <CurrentTaskParam name={"tag"}
                       label={"Tag"}
                       value={tag}
                       onChange={this.setParam}
                />
                <CurrentTaskParam name={"description"}
                       label={"Description"}
                       value={description}
                       onChange={this.setParam}
                />
                <CurrentTaskParam name={"numberOfPomidors"}
                       label={"Number of pomidors"}
                       value={numberOfPomidors}
                       onChange={this.setParam}
                />
                <button type="button" className="btn btn-success" onClick={this.add}>
                    Add
                </button>
            </div>
        )
    }
}

export default AddCurrentTask;