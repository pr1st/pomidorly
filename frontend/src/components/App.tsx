import React from 'react'
import '../css/App.css'

const App: React.FC = () => {
  return (
    <div className="App">
      YO
    </div>
  )
};

export default App;
