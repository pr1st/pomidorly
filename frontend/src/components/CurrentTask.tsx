import React from 'react';
import {CurrentTaskState} from "../types/currentTasks";

export type StateProps = CurrentTaskState

export type DispatchProps = {
    updateTask: (updateTask: CurrentTaskState) => void,
    deleteTask: (id: number) => void,
    swapTask: (id: number, isMoveUp: boolean) => void,
}

const CurrentTask = ({ id, tag, description, numberOfPomidors, inQueue, deleteTask, swapTask, updateTask}: StateProps & DispatchProps) => (
    <tr>
        <td>{tag}</td>
        <td>{description}</td>
        <td>{numberOfPomidors}</td>
        <td>
            <button className="btn-success btn-sm" onClick={e => {
                swapTask(id, true)
            }}>
                ⮝
            </button>
            <button className="btn-success btn-sm" onClick={e => {
                swapTask(id, false)
            }}>
                ⮟
            </button>
        </td>
        <td>updateHere</td>
        <td>deleteHere</td>
    </tr>
);

export default CurrentTask;