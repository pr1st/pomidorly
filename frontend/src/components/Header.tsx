import React from 'react';
import "bootstrap/dist/css/bootstrap.css"
import classNames from "classnames";

type Props = {
    toMain: () => void,
    toStatistics: () => void,
    toSignUp: () => void,
    toSignIn: () => void,
    userName: string,
    logOut: () => void
}

const Header = (props: Props) => {
    const {toMain, toStatistics, toSignIn, toSignUp, userName, logOut} = props;
    const btnClass = classNames("btn", "btn-success", "btn-lg");
    let login;
    if (userName === "") {
        login = (
            <div className={classNames("Login")}>
                <button className={btnClass}
                        onClick={toSignIn}>
                    Sign in
                </button>
                <button className={btnClass}
                        onClick={toSignUp}>
                    Sign up
                </button>
            </div>
        )
    } else {
        login = (
            <div className={classNames("Login")}>
                <p>
                    {userName}
                </p>
                <button className={btnClass}
                        onClick={logOut}>
                    Log out
                </button>
            </div>
        )
    }
    return (
        <div className={classNames("Header")}>
            <div className={classNames("Page")}>
                <button className={btnClass}
                        onClick={toMain}
                        disabled={userName === ""}>
                    Main
                </button>
                <button className={btnClass}
                        onClick={toStatistics}
                        disabled={userName === ""}>
                    Statistics
                </button>
            </div>
            {login}
        </div>
    )
};

export default Header;