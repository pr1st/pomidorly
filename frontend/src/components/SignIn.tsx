import React from 'react';

import "bootstrap/dist/css/bootstrap.css"

class SignIn extends React.Component<{},{}> {
    render() {
        return (
            <div>
                SignIn
            </div>
        );
    }
}

export default SignIn