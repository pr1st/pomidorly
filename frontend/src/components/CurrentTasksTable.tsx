import React from 'react';
import {CurrentTasksState, CurrentTaskState} from "../types/currentTasks";
import CurrentTask from "./CurrentTask";
import classNames from "classnames";
import AddCurrentTask from "./AddCurrentTask";

export type StateProps = {
    tasks: CurrentTasksState
};

export type DispatchProps = {
    updateTask: (updateTask: CurrentTaskState) => void,
    deleteTask: (id: number) => void,
    swapTask: (id1: number, id2: number) => void,
    addTask: (tag: string, description: string, numberOfPomidors: number) => void
}

const CurrentTasksTable = ({updateTask, swapTask, deleteTask, addTask, tasks}: StateProps & DispatchProps) => {
    const sortedTasks = tasks.sort((a, b) => a.inQueue - b.inQueue);
    return (
        <div className={classNames("CurrentTasks")}>
            <h1>Current Tasks</h1>
            <AddCurrentTask addTask={addTask}/>
            <table className="table table-bordered table-responsive-md table-striped">
                <tbody>
                <tr>
                    <th>Tag</th>
                    <th>Description</th>
                    <th>Pomidors</th>
                    <th>Sort</th>
                    <th>Update</th>
                    <th>Remove</th>
                </tr>
                {sortedTasks
                    .map((task, index) => {
                        const {tag, description, numberOfPomidors, id, inQueue} = task;
                        return (
                            <CurrentTask
                                id={id}
                                tag={tag}
                                description={description}
                                numberOfPomidors={numberOfPomidors}
                                inQueue={inQueue}
                                deleteTask={deleteTask}
                                updateTask={updateTask}
                                swapTask={(taskId, isMoveUp) => {
                                    if (isMoveUp && (index - 1 >= 0)) {
                                        swapTask(taskId, sortedTasks[index - 1].id)
                                    } else if (!isMoveUp && (index + 1 < sortedTasks.length)) {
                                        swapTask(taskId, sortedTasks[index + 1].id)
                                    }
                                }}
                                key={id}
                            />
                        )
                    })}
                </tbody>
            </table>
        </div>
    );
};

export default CurrentTasksTable;