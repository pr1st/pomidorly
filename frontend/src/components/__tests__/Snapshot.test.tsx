import React from 'react';
import renderer from 'react-test-renderer';
import Header from "../Header";
import AddCurrentTask from "../AddCurrentTask";
import App from "../App";
import {MAIN_PAGE} from "../../types/currentPage";
import CurrentTask from "../CurrentTask";
import CurrentTasksTable from "../CurrentTasksTable";
import HistoryTask from "../HistoryTask";
import HistoryTasksTable from "../HistoryTasksTable";
import SignIn from "../SignIn";
import SignUp from "../SignUp";
import Statistics from "../Statistics";
import Timer from "../Timer";
import TimerConfigDialog from "../TimerConfigDialog";
import UpdateCurrentTaskDialog from "../UpdateCurrentTaskDialog";

describe("Component snapshot test", () => {

    const dummy = () => {};

    it("AddCurrentTask", () => {
        const tree = renderer
            .create(
                <AddCurrentTask
                    addTask={dummy}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("CurrentTask", () => {
        const tree = renderer
            .create(
                <CurrentTask
                    deleteTask={dummy}
                    description={"DESC"}
                    id={2}
                    inQueue={3}
                    numberOfPomidors={123}
                    swapTask={dummy}
                    tag={"TAG"}
                    updateTask={dummy}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("CurrentTasksTable", () => {
        const tree = renderer
            .create(
                <CurrentTasksTable
                    updateTask={dummy}
                    swapTask={dummy}
                    deleteTask={dummy}
                    addTask={dummy}
                    tasks={[
                        {
                            id:1,
                            numberOfPomidors:123,
                            inQueue:3,
                            description:"ASDASD",
                            tag:"ASD"
                        }, {
                            id:2,
                            numberOfPomidors:1233,
                            inQueue:34,
                            description:"ASDASDAS",
                            tag:"ASDF"
                        }, {
                            id:5,
                            numberOfPomidors:123312,
                            inQueue:2,
                            description:"ds",
                            tag:"d"
                        }
                    ]}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Header", () => {
        const tree = renderer
            .create(
                <Header
                    logOut={dummy}
                    toMain={dummy}
                    toSignIn={dummy}
                    toSignUp={dummy}
                    toStatistics={dummy}
                    userName={"CustomUser"}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("HistoryTask", () => {
        const tree = renderer
            .create(
                <HistoryTask
                    deleteTask={dummy}
                    tag={"ASD"}
                    id={3}
                    description={"AAA"}
                    repeatTask={dummy}
                    timeFinished={111}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("HistoryTasksTable", () => {
        const tree = renderer
            .create(
                <HistoryTasksTable
                    repeatTask={dummy}
                    deleteTask={dummy}
                    tasks={[
                        {
                            id:1,
                            description:"ASDASDdd",
                            tag:"ASDd",
                            timeFinished:123
                        }, {
                            id:2,
                            description:"ASDASDdd",
                            tag:"ASDd",
                            timeFinished:123
                        }, {
                            id:5,
                            description:"ASDASDdd",
                            tag:"ASDd",
                            timeFinished:123
                        }
                    ]}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("SignIn", () => {
        const tree = renderer
            .create(
                <SignIn
                    setErrorMessage={dummy}
                    signIn={dummy}
                    errorMessage={"Error"}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("SignUp", () => {
        const tree = renderer
            .create(
                <SignUp
                    setErrorMessage={dummy}
                    signUp={dummy}
                    errorMessage={"Err"}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Statistics", () => {
        const tree = renderer
            .create(
                <Statistics
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Timer", () => {
        const tree = renderer
            .create(
                <Timer
                    doPomidor={dummy}
                    pauseTimer={dummy}
                    setConfig={dummy}
                    skipTimer={dummy}
                    startTimer={dummy}
                    stopTimer={dummy}
                    config={{
                        shortBreakDuration:1,
                        pomidorDuration:2,
                        alarmWhenZero:false,
                        longBreakDuration:3,
                        numberOfPomidorsBeforeLongBreak:5
                    }}
                    currentState={{
                        isActive:false,
                        currentPomidor:2,
                        startTime:123,
                        timeRemaining:3000,
                        isBreak:false
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("TimerConfigDialog", () => {
        const tree = renderer
            .create(
                <TimerConfigDialog
                    setConfig={dummy}
                    alarmWhenZero={false}
                    longBreakDuration={2}
                    numberOfPomidorsBeforeLongBreak={3}
                    pomidorDuration={5}
                    shortBreakDuration={123}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("UpdateCurrentTaskDialog", () => {
        const tree = renderer
            .create(
                <UpdateCurrentTaskDialog
                    updateTask={dummy}
                    id={1}
                    description={"ASD"}
                    tag={"ADSD"}
                    numberOfPomidors={2}
                    inQueue={3}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});