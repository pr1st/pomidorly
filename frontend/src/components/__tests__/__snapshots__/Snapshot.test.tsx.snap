// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot test AddCurrentTask 1`] = `
<div
  className="row AddCurrentTask"
>
  <label
    className="Param"
  >
    Tag
    :
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    className="Param"
  >
    Description
    :
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    className="Param"
  >
    Number of pomidors
    :
    <input
      className="form-control"
      onChange={[Function]}
      type="number"
      value={1}
    />
  </label>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</div>
`;

exports[`Component snapshot test CurrentTask 1`] = `
<tr>
  <td>
    TAG
  </td>
  <td>
    DESC
  </td>
  <td>
    123
  </td>
  <td>
    <button
      className="btn btn-outline-dark"
      onClick={[Function]}
    >
      ⮝
    </button>
    <button
      className="btn btn-outline-dark"
      onClick={[Function]}
    >
      ⮟
    </button>
  </td>
  <td>
    <button
      className="btn btn-outline-warning"
      data-backdrop="static"
      data-target="#updateCurrentTaskModal-2"
      data-toggle="modal"
    >
      Update
    </button>
    <div
      className="modal fade"
      id="updateCurrentTaskModal-2"
      tabIndex={-1}
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
            >
              Update task
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <label>
                Id: 
                2
              </label>
              <label
                className="Param"
              >
                Tag
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="text"
                  value="TAG"
                />
              </label>
              <label
                className="Param"
              >
                Description
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="text"
                  value="DESC"
                />
              </label>
              <label
                className="Param"
              >
                Number of pomidors
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={123}
                />
              </label>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Save changes
            </button>
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </td>
  <td>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
    >
      Remove
    </button>
  </td>
</tr>
`;

exports[`Component snapshot test CurrentTasksTable 1`] = `
<div
  className="CurrentTasks"
>
  <h1>
    Current Tasks
  </h1>
  <div
    className="row AddCurrentTask"
  >
    <label
      className="Param"
    >
      Tag
      :
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      className="Param"
    >
      Description
      :
      <input
        className="form-control"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label
      className="Param"
    >
      Number of pomidors
      :
      <input
        className="form-control"
        onChange={[Function]}
        type="number"
        value={1}
      />
    </label>
    <button
      className="btn btn-success"
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
  <table
    className="table table-bordered table-responsive-md table-striped"
  >
    <tbody>
      <tr>
        <th>
          Tag
        </th>
        <th>
          Description
        </th>
        <th>
          Pomidors
        </th>
        <th>
          Sort
        </th>
        <th>
          Update
        </th>
        <th>
          Remove
        </th>
      </tr>
      <tr>
        <td>
          d
        </td>
        <td>
          ds
        </td>
        <td>
          123312
        </td>
        <td>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮝
          </button>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮟
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-warning"
            data-backdrop="static"
            data-target="#updateCurrentTaskModal-5"
            data-toggle="modal"
          >
            Update
          </button>
          <div
            className="modal fade"
            id="updateCurrentTaskModal-5"
            tabIndex={-1}
          >
            <div
              className="modal-dialog"
            >
              <div
                className="modal-content"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                  >
                    Update task
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body"
                >
                  <div>
                    <label>
                      Id: 
                      5
                    </label>
                    <label
                      className="Param"
                    >
                      Tag
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="d"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Description
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="ds"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Number of pomidors
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="number"
                        value={123312}
                      />
                    </label>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <button
                    className="btn btn-primary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Save changes
                  </button>
                  <button
                    className="btn btn-secondary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
      <tr>
        <td>
          ASD
        </td>
        <td>
          ASDASD
        </td>
        <td>
          123
        </td>
        <td>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮝
          </button>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮟
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-warning"
            data-backdrop="static"
            data-target="#updateCurrentTaskModal-1"
            data-toggle="modal"
          >
            Update
          </button>
          <div
            className="modal fade"
            id="updateCurrentTaskModal-1"
            tabIndex={-1}
          >
            <div
              className="modal-dialog"
            >
              <div
                className="modal-content"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                  >
                    Update task
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body"
                >
                  <div>
                    <label>
                      Id: 
                      1
                    </label>
                    <label
                      className="Param"
                    >
                      Tag
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="ASD"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Description
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="ASDASD"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Number of pomidors
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="number"
                        value={123}
                      />
                    </label>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <button
                    className="btn btn-primary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Save changes
                  </button>
                  <button
                    className="btn btn-secondary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
      <tr>
        <td>
          ASDF
        </td>
        <td>
          ASDASDAS
        </td>
        <td>
          1233
        </td>
        <td>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮝
          </button>
          <button
            className="btn btn-outline-dark"
            onClick={[Function]}
          >
            ⮟
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-warning"
            data-backdrop="static"
            data-target="#updateCurrentTaskModal-2"
            data-toggle="modal"
          >
            Update
          </button>
          <div
            className="modal fade"
            id="updateCurrentTaskModal-2"
            tabIndex={-1}
          >
            <div
              className="modal-dialog"
            >
              <div
                className="modal-content"
              >
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                  >
                    Update task
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                </div>
                <div
                  className="modal-body"
                >
                  <div>
                    <label>
                      Id: 
                      2
                    </label>
                    <label
                      className="Param"
                    >
                      Tag
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="ASDF"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Description
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="text"
                        value="ASDASDAS"
                      />
                    </label>
                    <label
                      className="Param"
                    >
                      Number of pomidors
                      :
                      <input
                        className="form-control"
                        onChange={[Function]}
                        type="number"
                        value={1233}
                      />
                    </label>
                  </div>
                </div>
                <div
                  className="modal-footer"
                >
                  <button
                    className="btn btn-primary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Save changes
                  </button>
                  <button
                    className="btn btn-secondary"
                    data-dismiss="modal"
                    onClick={[Function]}
                    type="button"
                  >
                    Close
                  </button>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Component snapshot test Header 1`] = `
<div
  className="Header"
>
  <div
    className="Page"
  >
    <button
      className="btn btn-success btn-lg"
      disabled={false}
      onClick={[Function]}
    >
      Main
    </button>
    <button
      className="btn btn-success btn-lg"
      disabled={false}
      onClick={[Function]}
    >
      Statistics
    </button>
  </div>
  <div
    className="Login"
  >
    <p>
      CustomUser
    </p>
    <button
      className="btn btn-success btn-lg"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
</div>
`;

exports[`Component snapshot test HistoryTask 1`] = `
<tr>
  <td>
    ASD
  </td>
  <td>
    AAA
  </td>
  <td>
    7:00:00 AM
  </td>
  <td>
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Repeat
    </button>
  </td>
  <td>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
    >
      Remove
    </button>
  </td>
</tr>
`;

exports[`Component snapshot test HistoryTasksTable 1`] = `
<div
  className="HistoryTasks"
>
  <h1>
    History Tasks
  </h1>
  <table
    className="table table-bordered table-responsive-md table-striped"
  >
    <tbody>
      <tr>
        <th>
          Tag
        </th>
        <th>
          Description
        </th>
        <th>
          Time Finished
        </th>
        <th>
          Repeat
        </th>
        <th>
          Remove
        </th>
      </tr>
      <tr>
        <td>
          ASDd
        </td>
        <td>
          ASDASDdd
        </td>
        <td>
          7:00:00 AM
        </td>
        <td>
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Repeat
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
      <tr>
        <td>
          ASDd
        </td>
        <td>
          ASDASDdd
        </td>
        <td>
          7:00:00 AM
        </td>
        <td>
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Repeat
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
      <tr>
        <td>
          ASDd
        </td>
        <td>
          ASDASDdd
        </td>
        <td>
          7:00:00 AM
        </td>
        <td>
          <button
            className="btn btn-outline-primary"
            onClick={[Function]}
          >
            Repeat
          </button>
        </td>
        <td>
          <button
            className="btn btn-outline-danger"
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Component snapshot test SignIn 1`] = `
<div
  className="SignIn"
  onKeyDown={[Function]}
>
  <label>
    Name
    <input
      className="form-control"
      onChange={[Function]}
      value=""
    />
  </label>
  <label>
    Password
    <input
      className="form-control"
      onChange={[Function]}
      value=""
    />
  </label>
  <button
    className="btn btn-primary btn-lg"
    onClick={[Function]}
  >
    Sign in
  </button>
  <h5
    className="Error"
  >
    Error
  </h5>
</div>
`;

exports[`Component snapshot test SignUp 1`] = `
<div
  className="SignUp"
  onKeyDown={[Function]}
>
  <label>
    Name
    <input
      className="form-control"
      onChange={[Function]}
      value=""
    />
  </label>
  <label>
    Password
    <input
      className="form-control"
      onChange={[Function]}
      value=""
    />
  </label>
  <label>
    Password again
    <input
      className="form-control"
      onChange={[Function]}
      value=""
    />
  </label>
  <button
    className="btn btn-primary btn-lg"
    onClick={[Function]}
  >
    Sign up
  </button>
  <h5
    className="Error"
  >
    Err
  </h5>
</div>
`;

exports[`Component snapshot test Statistics 1`] = `
<div
  className="Statistics"
>
  Statistics
</div>
`;

exports[`Component snapshot test Timer 1`] = `
<div
  className="Timer"
>
  <div
    className="Options"
  >
    <button
      className="btn btn-outline-primary btn-lg"
      data-backdrop="static"
      data-target="#timerConfigModal"
      data-toggle="modal"
    >
      Config
    </button>
    <div
      className="modal fade"
      id="timerConfigModal"
      tabIndex={-1}
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header"
          >
            <h5
              className="modal-title"
            >
              Configuration properties
            </h5>
            <button
              aria-label="Close"
              className="close"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <div>
              <label>
                Pomidor duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={2}
                />
              </label>
            </div>
            <div>
              <label>
                Short break duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={1}
                />
              </label>
            </div>
            <div>
              <label>
                Long break duration
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={3}
                />
              </label>
            </div>
            <div>
              <label>
                Number of pomidors before long break
                :
                <input
                  className="form-control"
                  onChange={[Function]}
                  type="number"
                  value={5}
                />
              </label>
            </div>
            <div>
              <label>
                Alarm when zero
                <select
                  className="form-control"
                  onChange={[Function]}
                  value="false"
                >
                  <option
                    value="true"
                  >
                    true
                  </option>
                  <option
                    value="false"
                  >
                    false
                  </option>
                </select>
              </label>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-primary"
              data-dismiss="modal"
              onClick={[Function]}
              type="button"
            >
              Save changes
            </button>
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              id="closeDialog"
              onClick={[Function]}
              type="button"
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Time Pause"
  >
    0
    :
    03
  </div>
  <h5>
    Pause
  </h5>
  <p>
    Pomidors: 
    2
    /
    5
  </p>
  <div>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Start
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Pause
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Stop
    </button>
    <button
      className="btn btn-outline-primary btn-lg"
      onClick={[Function]}
    >
      Skip
    </button>
  </div>
</div>
`;

exports[`Component snapshot test TimerConfigDialog 1`] = `
<div
  className="modal fade"
  id="timerConfigModal"
  tabIndex={-1}
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Configuration properties
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          <label>
            Pomidor duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={5}
            />
          </label>
        </div>
        <div>
          <label>
            Short break duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={123}
            />
          </label>
        </div>
        <div>
          <label>
            Long break duration
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={2}
            />
          </label>
        </div>
        <div>
          <label>
            Number of pomidors before long break
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={3}
            />
          </label>
        </div>
        <div>
          <label>
            Alarm when zero
            <select
              className="form-control"
              onChange={[Function]}
              value="false"
            >
              <option
                value="true"
              >
                true
              </option>
              <option
                value="false"
              >
                false
              </option>
            </select>
          </label>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Save changes
        </button>
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          id="closeDialog"
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component snapshot test UpdateCurrentTaskDialog 1`] = `
<div
  className="modal fade"
  id="updateCurrentTaskModal-1"
  tabIndex={-1}
>
  <div
    className="modal-dialog"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Update task
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div>
          <label>
            Id: 
            1
          </label>
          <label
            className="Param"
          >
            Tag
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="text"
              value="ADSD"
            />
          </label>
          <label
            className="Param"
          >
            Description
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="text"
              value="ASD"
            />
          </label>
          <label
            className="Param"
          >
            Number of pomidors
            :
            <input
              className="form-control"
              onChange={[Function]}
              type="number"
              value={2}
            />
          </label>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Save changes
        </button>
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
